cli/internal/fs/fs.go

```diff
@@ -60,12 +68,33 @@ func FileExists(filename string) bool {
 	return err == nil && !info.IsDir()
 }
 
+// StatFile adds FileInfo to the given StatedFile if it doesn't already exist
+func StatFile(file StatedFile) (StatedFile, error) {
+	if file.Path == "" {
+		return file, errors.New("empty file path")
+	}
+	if file.Info == nil {
+		// we don't have file info already, look it up
+		fileInfos, err := os.Lstat(file.Path)
```

Comment:
Once `file.Path` is an `AbsolutePath` we have a lot of helper functions for things like `Lstat`.